Sloth.Web/Views/Transactions/Index.cshtml

```diff
@@ -7,17 +7,93 @@
 <div class="container">
     <h2>@ViewBag.Title</h2>
 
-    <partial name="_TransactionsTable" model="Model" />
+    <form id="filter" action="@Url.Action("Index")" onsubmit="return trySubmitFilter()">
```

Comment:
Instead of doing a submit each time the date value is changed, we usually just create a submit form/button for once they have changed the values.
The way you have it now, you can't change the to value to earlier than the from, so you would need to make the from (maybe to the previous year), get the data, then change to to value and get the data again.
If you change it so it doesn't do a bad request when the to is before the from, and just pass along the values it could work the way you have it.

Reply:
Okay. I was trying to keep the behavior consistent with the text search without flooding the back end with intermediate values. I'll make it update to a range of a month when the user enters an invalid range. Maybe I should do that in the controller instead of javascript in lieu of returning an error message?